city_metrix/layers/smart_cities_lulc.py

```diff
@@ -11,11 +23,194 @@ def __init__(self, land_cover_class=None, **kwargs):
+        esa_world_cover = EsaWorldCover().get_data(bbox)
 
+        reclass_map = {
+            EsaWorldCoverClass.TREE_COVER.value: 1,
+            EsaWorldCoverClass.SHRUBLAND.value: 1,
+            EsaWorldCoverClass.GRASSLAND.value: 1,
+            EsaWorldCoverClass.CROPLAND.value: 1,
+            EsaWorldCoverClass.BUILT_UP.value: 2,
+            EsaWorldCoverClass.BARE_OR_SPARSE_VEGETATION.value: 3,
+            EsaWorldCoverClass.SNOW_AND_ICE.value: 4,
+            EsaWorldCoverClass.PERMANENT_WATER_BODIES.value: 4,
+            EsaWorldCoverClass.HERBACEOUS_WET_LAND.value: 4,
+            EsaWorldCoverClass.MANGROVES.value: 4,
+            EsaWorldCoverClass.MOSS_AND_LICHEN.value: 3
+            # Add other mappings as needed
+        }
+        reclassified_esa = xr.apply_ufunc(
+            np.vectorize(lambda x: reclass_map.get(x, x)),
+            esa_world_cover,
+            vectorize=True
+        )
 
+        reclassified_esa = reclassified_esa.rio.write_crs(esa_world_cover.rio.crs, inplace=True)
 
+        esa_1m = reclassified_esa.rio.reproject(
+            dst_crs=crs,
+            resolution=1,
+            resampling=Resampling.nearest
+        )
 
 
+        # OSM tags
+        open_space_tag = {'leisure': ['park', 'nature_reserve', 'common', 'playground', 'pitch', 'track', 'garden', 'golf_course', 'dog_park', 'recreation_ground', 'disc_golf_course'],
+                          'boundary': ['protected_area', 'national_park', 'forest_compartment', 'forest']}
+        water_tag = {'water': True,
+                     'natural': ['water'],
+                     'waterway': True}
+        roads_tag = {'highway': ["residential", "service", "unclassified", "tertiary", "secondary", "primary", "turning_circle", "living_street", "trunk", "motorway", "motorway_link", "trunk_link",
+                                 "primary_link", "secondary_link", "tertiary_link", "motorway_junction", "turning_loop", "road", "mini_roundabout", "passing_place", "busway"]}
+        building_tag = {'building': True}
+        parking_tag = {'amenity': ['parking'],
+                       'parking': True}
 
+        def rasterize_osm(gdf, snap_to):
+            raster = make_geocube(
+                vector_data=gdf,
+                measurements=["Value"],
+                like=snap_to,
+                fill=0
+            ).Value
+
+            return raster.rio.reproject_match(snap_to)
+
+
+        # Open space
+        open_space_osm = OpenStreetMap(osm_tag=open_space_tag).get_data(bbox).to_crs(crs).reset_index()
+        open_space_osm['Value'] = 10
+        open_space_1m = rasterize_osm(open_space_osm, esa_1m)
+
+
+        # Water
+        water_osm = OpenStreetMap(osm_tag=water_tag).get_data(bbox).to_crs(crs).reset_index()
+        water_osm['Value'] = 20
+        water_1m = rasterize_osm(water_osm, esa_1m)
+
+
+        # Roads
+        roads_osm = OpenStreetMap(osm_tag=roads_tag).get_data(bbox).to_crs(crs).reset_index()
```

Comment:
for these last 3, perhaps make it 3 subclasses of OSM that each just pass these tags? I think there are other indicators we might to re-use these for, and would simplify this very long function

Reply:
Updated OSM layer to have predefined land use types to be reused